docs/source/howto/workflows.rst

```diff
@@ -4,14 +4,217 @@
+
+        def validate_result(self):
+
+            result = self.ctx['addition'].outputs.sum
+
+            if result.value < 0:
+                return self.exit_codes.ERROR_NEGATIVE_NUMBER
+
+        def result(self):
+            self.out('result', self.ctx['addition'].outputs.sum)
+
+You can give the work chain any valid Python class name, but the convention is to have it end in ``WorkChain`` so that it is always immediately clear what it references.
+Let's go over the methods of the ``MultiplyAddWorkChain`` one by one:
+
+.. code-block:: python
+
+    @classmethod
+    def define(cls, spec):
+        """Specify inputs and outputs."""
+        super(MultiplyAddWorkChain, cls).define(spec)
+        spec.input('x', valid_type=Int)
+        spec.input('y', valid_type=Int)
+        spec.input('z', valid_type=Int)
+        spec.input('code', valid_type=Code)
+        spec.outline(
+            cls.multiply,
+            cls.add,
+            cls.validate_result,
+            cls.result
+        )
+        spec.output('result', valid_type=Int)
+        spec.exit_code(400, 'ERROR_NEGATIVE_NUMBER',
+            message='The result is a negative number.')
+
+The most important method to implement for every work chain is the ``define()`` method.
+This class method must always start by calling the ``define()`` method of its parent class.
+Next, the ``define()`` method allows the developer to define the characteristics of the work chain, which are contained in the work chain ``spec``:
+
+* the **inputs**, specified using the ``spec.input()`` method.
+  The first argument of the ``input()`` method is a string that specifies the label of the input, e.g. ``'x'``.
+  The ``valid_type`` keyword argument allows you to specify the required node type of the input.
+  Other keyword arguments allow the developer to set a default for the input, or indicate that an input should not be stored in the database, see :ref:`the process topics section <topics:processes:usage:spec>` for more details.
+* the **outline** or logic of the workflow, specified using the ``spec.outline()`` method.
+  The outline of the workflow is constructed from the methods of the ``WorkChain`` class.
+  For the ``MultiplyAddWorkChain``, the outline is a simple linear sequence of steps, but it's possible to define more complex workflows as well.
+  See the :ref:`work chain outline section <topics:workflows:usage:workchains:define_outline>` for more details.
+* the **outputs**, specified using the ``spec.output()`` method.
+  This method is very similar in its usage to the ``input()`` method.
+* the **exit codes** of the work chain, specified using the ``spec.exit_code()`` method.
+  Exit codes are used to clearly communicate known failure modes of the work chain to the user.
+  The first and second arguments define the ``exit_status`` of the work chain in case of failure (``400``) and the string that the developer can use to reference to the exit code (``ERROR_NEGATIVE_NUMBER``).
```

Comment:
```suggestion
  The first and second arguments define the ``exit_status`` of the work chain in case of failure (``400``) and the string that the developer can use to reference the exit code (``ERROR_NEGATIVE_NUMBER``).
```